pycernan/avro/base_client.py

```diff
@@ -30,12 +34,18 @@ def _recv_exact(self, sock, n_bytes):
 
     def _send(self, payload_id, sync, payload):
         with self.pool.connection() as sock:
+            metrics.publish_count.inc()
             self._send_exact(sock, payload)
             if sync:
-                self._wait_for_ack(sock, payload_id)
+                metrics.ack_request_count.inc()
+                with metrics.ack_latency():
+                    self._wait_for_ack(sock, payload_id)
+                metrics.ack_count.inc()
 
     def _wait_for_ack(self, sock, payload_id):
-        id_bytes = self._recv_exact(sock, 8)
+        id_bytes = self._recv_exact(sock, NUM_ID_BYTES)
+        metrics.bytes_received.inc(len(id_bytes))
         (recv_id,) = struct.unpack(">Q", id_bytes)
         if recv_id != payload_id:
+            metrics.ack_invalid_count.inc()
```

Comment:
Why don't we catch all exceptions at a top level and set a label for the exception type ? This way we support tracking all exceptions.

Reply:
We will do that independently of this, considering we check this directly may as well increment the counter here.